Assets/Scripts/SelectedCellManager.cs

```diff
@@ -0,0 +1,188 @@
+    {
+        if (onPhotoAdded != null)
+        {
+            onPhotoAdded(photoPaths);
+        }
+    }
+
+    private void TriggerPhotoDeleted(List<string> photoPaths)
+    {
+        if (onPhotoDeleted != null)
+        {
+            onPhotoDeleted(photoPaths);
+        }
+    }
+    //______________________________________________________________________________________
+    //SELECTED CELL && POP UP EVENTS 
+    //______________________________________________________________________________________
+    private void TryAddCellData()
+    {
+        if (!StateSaver.Data.Contains(selectedDayCell.DaycellData))
+        {
+            StateSaver.Data.Add(selectedDayCell.DaycellData);
+
+        }
+
+
+    }
+
+
+    private void HandleSelectCell(Cell cell)
+    {
+
+        if (pop == null)
+        {
+            selectedDayCell = cell as DayCell;
+
+            pop = Instantiate(popUpPrefab, canvas);
+            pop.MakeButton(stickerButtonName, HandleAddStickerPressed, false);
+            pop.MakeButton(notesButtonName, HandleAddContentPressed, false);
+            pop.SetPosition(cell.transform.position);
+        }
+
+    }
+    private void HandleQuitPopUp()
+    {
+
+        selectedDayCell = null;
+        Destroy(pop.gameObject);
+        pop = null;
+    }
+
+    //______________________________________________________________________________________
+    //SELECTED CELL STICKER EVENTS
+    //______________________________________________________________________________________
+    private void HandleAddStickerPressed()
+    {
+
+        pop.DestroyButons();
+        pop.MakeButton(poopButtonName, HandleAddPoopPressed, selectedDayCell.DaycellData.isPoopImageActive);
+        pop.MakeButton(pillButtonName, HandleAddMedicinePressed, selectedDayCell.DaycellData.isMedicineImageActive);
+
+
+    }
+
+    private void HandleAddPoopPressed()
+    {
+        selectedDayCell.SetPoopImage();
+        TryAddCellData();
+    }
+
+    private void HandleAddMedicinePressed()
+    {
+        selectedDayCell.SetMedicineImage();
+        TryAddCellData();
+    }
+
+    //______________________________________________________________________________________
+    //      CONTENT && DESCRIPTION EVENTS
+    //______________________________________________________________________________________
+    private void HandleAddContentPressed()
+    {
+
+        if (selectedCellContentWindow == null)
+        {
+            selectedCellContentWindow = Instantiate(selectedCellContentWindowPrefab, canvas);
+            selectedCellContentWindow.Configure(selectedDayCell.DaycellData.description, selectedDayCell.DaycellData.photoPaths);
+            tempPhotoPaths = new List<string>(selectedDayCell.DaycellData.photoPaths);
+        }
+
+
+    }
+    private void HandleSetNoteText(string str)
+    {
+        tempNote = str;
+    }
+    private void HandleCancelChanges()
+    {
+        tempNote = "";
```

Comment:
this is fine i do it all the time, some people say string.Empty is better, and so I try to go for that when I can, but I don't know why in all honesty

Reply:
Resolved